crates/plugins/mediator-coordination/src/didcomm/bridge.rs

```diff
@@ -36,28 +37,53 @@ impl DIDResolver for LocalDIDResolver {
             return Ok(Some(self.diddoc.clone()));
         }
 
-        if !did.starts_with("did:key:") {
-            return Err(Error::new(
+        if did.starts_with("did:peer:") {
+            // Adding logic to resolve Peer DIDs
+            let method = DidPeer::new_with_format(PublicKeyFormat::Jwk); // Assuming a Peer DID resolver
+            match method.expand(did) {
+                Ok(diddoc) => {
+                    // Attempt to convert DID document representation, propagating errors
+                    match serde_json::from_value(json!(Document {
+                        service: Some(vec![]),
+                        ..diddoc
+                    })) {
+                        Ok(doc) => Ok(Some(doc)),
+                        Err(err) => Err(Error::new(
+                            ErrorKind::DIDNotResolved, 
+                            Box::new(err)
+                        )),
+                    }
+                },
+                Err(err) => Err(Error::new(ErrorKind::DIDNotResolved, Box::new(err))),
```

Comment:
Why boxing the error here? You can simply return the error.
```rust
Err(err) => Err(Error::new(ErrorKind::DIDNotResolved, err)),
```
And since both resolutions now return the same error, you don't need to handle them separately.